Menus/MainMenu/MainMenu.cs

```diff
@@ -143,77 +158,130 @@ void CreateFaceButtons(List<Type> types)
 			foreach (var type in types)
 			{
 				var customMenuAttribute = (MainMenuItemAttribute)type.GetCustomAttributes(typeof(MainMenuItemAttribute), false).FirstOrDefault();
-				if (customMenuAttribute != null && !customMenuAttribute.shown)
-					continue;
-
-				var isTool = typeof(ITool).IsAssignableFrom(type);
+				var isTool = typeof(ITool).IsAssignableFrom(type) && menuTools.Contains(type);
 				var isWorkspace = typeof(Workspace).IsAssignableFrom(type);
 				var isSettingsProvider = typeof(ISettingsMenuProvider).IsAssignableFrom(type);
+				var isSettingsItemProvider = typeof(ISettingsMenuItemProvider).IsAssignableFrom(type);
 
-				var buttonData = new MainMenuUI.ButtonData();
-				buttonData.name = type.Name;
+				ITooltip tooltip = null;
+				MainMenuUI.ButtonData buttonData = null;
 
-				if (customMenuAttribute != null)
-				{
-					buttonData.name = customMenuAttribute.name;
-					buttonData.sectionName = customMenuAttribute.sectionName;
-					buttonData.description = customMenuAttribute.description;
-				}
-				else if (isTool)
+				var selectedType = type; // Local variable for closure
+				if (customMenuAttribute != null && customMenuAttribute.shown)
 				{
-					buttonData.name = type.Name.Replace("Tool", string.Empty);
+					tooltip = customMenuAttribute.tooltip;
+
+					buttonData = new MainMenuUI.ButtonData
+					{
+						name = customMenuAttribute.name,
+						sectionName = customMenuAttribute.sectionName,
+						description = customMenuAttribute.description
+					};
 				}
-				else if (isWorkspace)
+
+				if (isTool)
 				{
-					// For workspaces that haven't specified a custom attribute, do some menu categorization automatically
-					buttonData.name = type.Name.Replace("Workspace", string.Empty);
-					buttonData.sectionName = "Workspaces";
+					if (buttonData == null)
+						buttonData = new MainMenuUI.ButtonData { name = type.Name.Replace("Tool", string.Empty) };
+
+					CreateFaceButton(buttonData, tooltip, () =>
+					{
+						if (targetRayOrigin)
+						{
+							this.SelectTool(targetRayOrigin, selectedType);
+							UpdateToolButtons();
+						}
+					});
 				}
-				else if (isSettingsProvider)
+
+				if (isWorkspace)
```

Comment:
I'm not sure. Generally, a tool which is also a workspace would probably have an issue being either one independently. I actually haven't tested this to see what happens.

I also had a similar issue with the LocomotionTool in that it is a tool and a SettingsMenuItemProvider, but should not actually be included in the tools menu. Hence the added check to see if the tool is in the list of menuTools.

Generally, the reason for this change was because the LocomotionTool is now a type that is two things as far as the menu is concerned. We could alternatively go back to the previous setup and add a new class to delegate the ISettingsMenuItemProvider stuff.